### PR TITLE
#575 request req line 3

### LINKED ISSUE #575: Supplier requisition add/edit item modal
'Epic' issue for the supplier requisition add/edit item modal.

Really unsure about the current ui designs to break it down into tasks just yet

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #575 

- Gone with this for now
image
- The columns of the requisition aren't showing nicely, but will revisit that in another PR
- Note there is a bug with the API where you can't add a requisition line that is in another requisition, I believe - at the moment this won't actually give any feedback
- Also note that there is no `comment` field on the api update input, so comment changes won't persist

### CODE REVIEW

**Review by mark-prins — APPROVED** [review]
🎉  wow - looking awesome! nice to have requisitions coming along 😁 

You mentioned the columns - was going to say something about the SoH showing NaN - but expect that will be in a further PR.

File: packages/requisitions/src/RequestRequisition/api/hooks.ts [hunk 1]
```diff
@@ -140,7 +140,19 @@ export const useIsRequestRequisitionDisabled = (): boolean => {
...
 export const useSaveRequestLines = () => {
+  const { requisitionNumber = '' } = useParams();
   const { store } = useHostContext();
+  const queryClient = useQueryClient();
   const api = useRequestRequisitionApi();
-  return useMutation(RequestRequisitionQueries.upsertLine(api, store.id));
+
+  return useMutation(RequestRequisitionQueries.upsertLine(api, store.id), {
+    onSuccess: () => {
+      console.log(['requisition', store.id, requisitionNumber]);
```
[the remark this answers is not in the document]
Comment (joshxg) [reply packages/requisitions/src/RequestRequisition/api/hooks.ts]: I feel like we should make this an error in the linter for me 😁

**Comment by joshxg** [comment]
Thanks! think it's looking better? 

A small button 'use suggested quantity' might be useful to auto-fill the requested quantity, also.

image

**Comment by mark-prins** [comment]
looking very nice!! 😁 
one extra suggestion - perhaps make the modal wider to get the item autocomplete to show on the same line as the label?

**Comment by joshxg** [comment]
Do you mean "Stock Details"? That is more of a label a header for the column in the top part.. i.e. in line with `Comment` and `Order` .. looks a bit strange this way, but I think looks stranger the other way.. 😁

**Comment by mark-prins** [comment]
oooooooohh. 🤦 I see. byeee

**Pull request merged**